commentHandler.go

```diff
@@ -130,24 +134,84 @@ func handleComment(req types.IssueCommentOuter) {
 	case "close", "reopen":
 		fmt.Printf("%s wants to %s issue #%d\n", req.Comment.User.Login, command.Type, req.Issue.Number)
 
+		newState, validAction := assessState(command.Type, req.Issue.State)
+
+		if !validAction {
+			fmt.Printf("Request to %s issue #%d by %s was invalid.\n", command.Type, req.Issue.Number, req.Comment.User.Login)
+			return
+		}
+
 		client, ctx := makeClient(req.Installation.ID)
+		input := &github.IssueRequest{State: &newState}
+
+		_, _, err := client.Issues.Edit(ctx, req.Repository.Owner.Login, req.Repository.Name, req.Issue.Number, input)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		fmt.Printf("Request to %s issue #%d by %s was successful.\n", command.Type, req.Issue.Number, req.Comment.User.Login)
 
-		var state string
+		break
+
+	case "SetTitle":
+
+		fmt.Printf("%s wants to set the title of issue #%d\n", req.Comment.User.Login, req.Issue.Number)
+
+		newTitle := command.Value
 
-		if command.Type == "close" {
-			state = closed
-		} else if command.Type == "reopen" {
-			state = open
+		if newTitle == req.Issue.Title {
+			fmt.Printf("Setting the title of #%d by %s was unsuccessful as the new title was empty or unchanged.\n", req.Issue.Number, req.Comment.User.Login)
+			return
 		}
-		input := &github.IssueRequest{State: &state}
+
+		client, ctx := makeClient(req.Installation.ID)
+
+		input := &github.IssueRequest{Title: &newTitle}
 
 		_, _, err := client.Issues.Edit(ctx, req.Repository.Owner.Login, req.Repository.Name, req.Issue.Number, input)
 		if err != nil {
 			log.Fatalln(err)
 		}
-		fmt.Printf("Request to %s issue #%d by %s was successful.\n", command.Type, req.Issue.Number, req.Comment.User.Login)
+
+		fmt.Printf("Request to set the title of issue #%d by %s was successful.\n", req.Issue.Number, req.Comment.User.Login)
 
 		break
+
+	case "Lock":
+		fmt.Printf("%s wants to lock issue #%d\n", req.Comment.User.Login, req.Issue.Number)
```

Comment:
This code block is getting messy - are we at the point in this PR where we need new methods that optionally return `error` ?

Reply:
This was in my thinking as we move to packages

Reply:
Can we break out into functions first please